doc/api/http2.md

```diff
@@ -39,6 +39,12 @@ server.on('stream', (stream, headers) => {
 server.listen(80);
 ```
 
+Note that this is an HTTP server and does not support HTTPS.
```

Comment:
Or perhaps better yet: 'HTTP/2 server', which is used elsewhere in this document.